user-impl/src/test/java/com/example/auction/user/impl/UserRepositoryTest.java

```diff
@@ -56,8 +56,12 @@ public static void afterAll() {
     private final String email = "admin@gmail.com";
     private final String password = PUserCommand.hashPassword("admin");
     private final Instant createdAt = Instant.now();
-
+    int size = 25;
+    PaginatedSequence<User> createdUsers = Await.result(userRepository.getUsers(1, 10));
     private final PUser userCreated = new PUser(userId, createdAt, name, email, password);
+    int initialCount = createdUsers.getCount();
```

Comment:
I think it's better to have it inside the test method instead. 

`size` is the number of users you plan to create inside `shouldPaginateUserRetrieval` method. 
Preferably, we should use this value in the for loop as well.

```
for (int i = 0; i < size; i++) { ... }
```

The `initialCount` as well. This call will happen on class initialisation. At that point, the initial count is 0. 

My suggestion was more something like the following:

```
 @Test
public void shouldPaginateUserRetrieval() throws InterruptedException, ExecutionException, TimeoutException {

    PaginatedSequence<User> usersCreatedBefore = Await.result(userRepository.getUsers(1, 10));
    int initialCount = usersCreatedBefore.getCount();
    int size = 25;

    for (int i = 0; i < size; i++) {
        feed(new PUserEvent.PUserCreated(buildFixture(createdAt, i)));
    }

    PaginatedSequence<User> createdUsers = Await.result(userRepository.getUsers(1, 10));
    assertEquals(size + initialCount, createdUsers.getCount());
}
```